aws/data_source_aws_secretsmanager_secret.go

```diff
@@ -104,6 +110,23 @@ func dataSourceAwsSecretsManagerSecretRead(d *schema.ResourceData, meta interfac
 	d.Set("rotation_enabled", output.RotationEnabled)
 	d.Set("rotation_lambda_arn", output.RotationLambdaARN)
 
+	pIn := &secretsmanager.GetResourcePolicyInput{
+		SecretId: aws.String(d.Id()),
+	}
+	log.Printf("[DEBUG] Reading Secrets Manager Secret policy: %s", pIn)
+	pOut, err := conn.GetResourcePolicy(pIn)
+	if err != nil {
+		return fmt.Errorf("error reading Secrets Manager Secret policy: %s", err)
+	}
+
+	if pOut.ResourcePolicy != nil {
```

Comment:
While we are performing `nil` checks here, we should also ensure `pOut` is not `nil` as well to prevent the panic 😄 

```go
if pOut != nil && pOut.ResourcePolicy != nil {
```